codeserver/ubi9-python-3.11/Dockerfile.cpu

```diff
@@ -105,19 +118,30 @@ RUN sed -i -f ${NGINX_APP_ROOT}/nginxconf.sed ${NGINX_CONF_PATH} && \
     chown -R 1001:0 /opt/app-root/src/.config/code-server
 
 ## Configure nginx
-COPY ${SOURCE_CODE}/nginx/serverconf/ /opt/app-root/etc/nginx.default.d/
-COPY ${SOURCE_CODE}/nginx/httpconf/ /opt/app-root/etc/nginx.d/
-COPY ${SOURCE_CODE}/nginx/api/ /opt/app-root/api/
+COPY ${CODESERVER_SOURCE_CODE}/nginx/serverconf/ /opt/app-root/etc/nginx.default.d/
+COPY ${CODESERVER_SOURCE_CODE}/nginx/httpconf/ /opt/app-root/etc/nginx.d/
+COPY ${CODESERVER_SOURCE_CODE}/nginx/api/ /opt/app-root/api/
 
 # Launcher
-COPY --chown=1001:0 ${SOURCE_CODE}/run-code-server.sh ${SOURCE_CODE}/run-nginx.sh ./
+COPY --chown=1001:0 ${CODESERVER_SOURCE_CODE}/run-code-server.sh ${CODESERVER_SOURCE_CODE}/run-nginx.sh ./
 
-ENV SHELL /bin/bash
+ENV SHELL=/bin/bash
 
 ENV PYTHONPATH=/opt/app-root/bin/python3
 
-WORKDIR /opt/app-root/src
-
 USER 1001
 
+# Install usefull packages from Pipfile.lock
+COPY ${CODESERVER_SOURCE_CODE}/Pipfile.lock ./
+
+# Install packages and cleanup
+RUN echo "Installing softwares and packages" && \
+    micropipenv install && \
+    rm -f ./Pipfile.lock && \
+    # Fix permissions to support pip in Openshift environments \
+    chmod -R g+w /opt/app-root/lib/python3.11/site-packages && \
+    fix-permissions /opt/app-root -P
+
+WORKDIR /opt/app-root/src
+
 CMD ["/opt/app-root/bin/run-code-server.sh"]
```

Comment:
Eventually I think we should unify to use ENTRYPOINT for all our images in the future. Or do we have some reason to use ENTRYPOINT somewhere and CMD command elsewhere that I'm missing? :thinking: 

https://www.docker.com/blog/docker-best-practices-choosing-between-run-cmd-and-entrypoint/

Note: again, not something to deal with in this PR.

Reply:
It's that somewhere we're using the s2i entrypoint and somewhere we decided not to... We should stop using s2i also ;P